### PR TITLE
Feature/eui 670 user permissions

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/user-profile/services/user.service.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
+import { Observable, noop } from 'rxjs';
```
Comment (adnanakgun) [inline src/user-profile/services/user.service.ts]: Remove `noop` if it is not being used.

File: src/users/components/invite-user-permissions/invite-user-permission.component.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import {Component, Input, Output, EventEmitter} from '@angular/core';
+import { FormGroup } from '@angular/forms';
+
+@Component({
+    selector: 'app-invite-permission-form',
```
Comment (adnanakgun) [inline src/users/components/invite-user-permissions/invite-user-permission.component.ts]: Not the end of the world, but maybe the selector should be `app-invite-user-permission`

File: src/users/store/effects/invite-user.effects.ts [hunk 3]
```diff
@@ -39,4 +40,32 @@ export class InviteUserEffects {
     })
   );
 
+  @Effect()
```
Comment (adnanakgun) [inline src/users/store/effects/invite-user.effects.ts]: Please do these in `user.effects`

File: src/user-profile/services/user.service.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/user-profile/services/user.service.ts]: Done

File: src/users/containers/edit-user-permissions/edit-user-permission.component.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    }
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe(this.userSubscription);
+    this.unsubscribe(this.dependanciesSubscription);
+  }
+
+  onSubmit() {
+    if (this.editUserForm.valid) {
```
Comment (ficklephil) [inline src/users/containers/edit-user-permissions/edit-user-permission.component.ts]: Think about using a JS early return here, to get rid of the nesting.

File: src/users/store/effects/invite-user.effects.ts (hunk 3 above)
Comment (udaydenduluri33) [reply src/users/store/effects/invite-user.effects.ts]: Done

File: src/users/containers/utils/user-roles-util.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import { AppConstants } from '../../../app/app.constants';
```
Comment (ficklephil) [inline src/users/containers/utils/user-roles-util.ts]: No unit tests around these functions.

Comment (udaydenduluri33) [reply src/users/containers/utils/user-roles-util.ts]: Done

File: src/users/containers/edit-user-permissions/edit-user-permission.component.ts [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    ) { }
+
+    ngOnInit(): void {
+      this.isLoading$ = this.userStore.pipe(select(fromStore.getGetUserLoading));
+
+      this.dependanciesSubscription = combineLatest([
+      this.routerStore.pipe(select(fromRoot.getRouterState)),
+      this.userStore.pipe(select(fromStore.getGetUserLoaded))
+    ]).subscribe(([route, users]) => {
+      if (users === false) {
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply src/users/containers/edit-user-permissions/edit-user-permission.component.ts]: Done

File: src/users/containers/edit-user-permissions/edit-user-permission.component.ts (hunk 4 above)
Comment (udaydenduluri33) [reply src/users/containers/edit-user-permissions/edit-user-permission.component.ts]: Done

File: api/editUserPermissions/index.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        logger.info('error', error)
+        errReport = getErrorReport(getErrorMessage(error), error.status, getErrorMessage(error))
+        res.status(error.status).send(errReport)
+    }
+}
+
+function getErrorMessage(error: any): string {
+    return error && error.data ? error.data.message : ''
+}
+function getErrorReport(apiError: string, apiStatusCode: string, message: string): any {
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply api/editUserPermissions/index.ts]: Done

File: api/editUserPermissions/index.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+export const router = express.Router({ mergeParams: true })
+const logger = log4jui.getLogger('outgoing')
+
+router.put('', inviteUserRoute)
+
+async function inviteUserRoute(req, res) {
+    let errReport: any
+    if (!req.params.userId) {
+        errReport = getErrorReport('UserId is missing', '400', 'User Permissions route error')
+        res.status(500).send(errReport)
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply api/editUserPermissions/index.ts]: Done

File: api/editUserPermissions/index.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply api/editUserPermissions/index.ts]: Done, created an interface ErrorReport

**Pull request merged**